framework/wazuh/agent.py

```diff
@@ -1519,48 +1519,59 @@ def create_multi_group(group_id):
 
 
     @staticmethod
-    def remove_multi_group(group_id):
+    def remove_multi_group(groups_id):
+        """
+        Removes groups by IDs.
 
-        if group_id.lower() == "default":
-                raise WazuhException(1712)
+        :param groups_id: list with Groups ID.
+        """
 
-        for filename in listdir("{0}".format(common.groups_path)):
-            file = open("{0}/{1}".format(common.groups_path,filename),"r")
-            agent_id = filename
-            agent_group = file.read()
-            agent_group = agent_group.strip()
-            file.close()
+        if not isinstance(groups_id, list):
+            raise WazuhException(1104, "Input must be a list")
 
-            if agent_group.find(group_id) >= 0:
+        for group in groups_id:
+            if group.lower() == "default":
```

Comment:
Instead of iterating over all groups, use `in` operator. Also, `in` operator for lists is _O(n)_, so you may consider using a set instead of a list.